src/net/sourceforge/kolmafia/session/BastilleBattalionManager.java

```diff
@@ -928,4 +1147,111 @@ public static final boolean registerRequest(final String urlString) {
 
     return true;
   }
+
+  // *** Game Logging for Analysis
+
+  //
+  // Since I am not (yet) working on a script to automate this, I want the data to be automatically
+  // saved to a file in a format which can be read by an analysis script via file_to_map().
+  //
+  // This is the proposed format.
+  //
+  // The file is a tab delimited file in "data" named Bastille.battles.txt
+  //
+  // Note that this is not user specific; you and all of your multis will contribute to the results
+  //
+  // The "key" is DATE.PLAYERID.GAME.ROUND
+  //
+  // The fields are designed to be easily loaded into an ASH record.
+  //
+  // 20220409.ID.1.1	1	1	4	-1	6	0	5	MCP	bigcastle	offense	true	true	false	true	142
+  //
+  // record battle {
+  //     int number;         // Affects strength of enemy
+  //     int [6] stats;      // MA/MD/CA/CD/PA/PD
+  //     string boosts;      // MCP
+  //     string enemy;       // frenchcastle,masterofnone,bigcastle,
+  //                         // berserker,shieldmaster,barracks
+  //     string stance;      // {offensive,waiting,defensive}
+  //     boolean aggressor;  // as opposed to defender
+  //     boolean military;   // true if won
+  //     boolean castle;     // true if won
+  //     boolean psych;      // true if won
+  //     int cheese;         // if won
+  // };
+  //
+  // battle [string] battles;
+  // file_to_map( FILENAME, battles );
+
+  private static final String BATTLE_FILE_NAME = "Bastille.battles.txt";
+
+  private static final String TAB = "\t";
+
+  private static String generateKey(int game, int number) {
+    StringBuilder buf = new StringBuilder();
+    buf.append(KoLConstants.DAILY_FORMAT.format(new Date()));
+    buf.append(".");
+    buf.append(KoLCharacter.getPlayerId());
+    buf.append(".");
+    buf.append(game);
+    buf.append(".");
+    buf.append(number);
+    return buf.toString();
+  }
+
+  private static String generateFields(Battle battle) {
+    StringBuilder buf = new StringBuilder();
+    buf.append(battle.number);
```

Comment:
I would probably make a `String makeTabSeparated(String... fields)` method instead of manually putting a TAB between the fields.

Reply:
That is a fine idea. I extended it and it's a lot easier to read now.